source/brailleDisplayDrivers/braillePen.py

```diff
@@ -0,0 +1,143 @@
+				continue			
+			break
+		else:
+			raise RuntimeError("BraillePen not found")
+				
+	def terminate(self):
+		try:
+			super(BrailleDisplayDriver, self).terminate()
+		finally:
+			self._dev.close()
+		
+	def _serOnReceive(self,data):
+		braille=None
+		command=None
+		braille=data
+		command=self._dev.read(1)
+		self._serHandleResponse(braille, command)
+
+	def _serHandleResponse(self,braille,command):		
+		if braille is not None and command is not None:
+			try:
+				inputCore.manager.executeGesture(InputGesture(braille, command))
+			except inputCore.NoInputGestureAction:
+				pass
+	
+	def display(self, cells):
+		pass
+
+	gestureMap = inputCore.GlobalGestureMap({
+		"globalCommands.GlobalCommands": {
+			"kb:upArrow": ("br(braillePen):space+dot1",),
+			"kb:downArrow": ("br(braillePen):space+dot4",),
+			"kb:leftArrow": ("br(braillePen):space+dot3",),
+			"kb:rightArrow": ("br(braillePen):space+dot6",),
+			"kb:enter": ("br(braillePen):space+dot8"),			
+			"kb:backspace": ("br(braillePen):space+dot7","br(braillePen):space+dot1+dot2"),
+			"kb:delete": ("br(braillePen):space+dot1+dot4+dot5",),
+			"kb:windows": ("br(braillePen):space+dot3+dot4",),
+			"kb:alt": ("br(braillePen):space+dot1+dot3+dot4",),
+			"kb:escape": ("br(braillePen):space+dot1+dot5",),
+			"kb:shift+tab": ("br(braillePen):space+dot2+dot3+dot4+dot5+dot8",),
+			"kb:tab": ("br(braillePen):space+dot2+dot3+dot4+dot5+dot7",),			
+			"toggleInputHelp": ("br(braillePen):space+dot3+dot6+dot7",),
+			"braille_toggleTether": ("br(braillePen):space+dot1+dot2+dot6",),
+			"kb:home": ("br(braillePen):space+dot2+dot3",),
+			"kb:end": ("br(braillePen):space+dot5+dot6",),
+			"sayAll": ("br(braillePen):space+dot1+dot2+dot4+dot5+dot6",),
+			"dateTime": ("br(braillePen):space+dot2+dot3+dot4",),
+			"reportCurrentSelection": ("br(braillePen):space+dot1+dot2+dot4+dot5+dot6+dot7",),
+			"showGui": ("br(braillePen):space+dot1+dot2+dot4+dot5",),
+			"kb:alt+tab": ("br(braillePen):space+dot1+dot3+dot5",),
+			"say_battery_status": ("br(braillePen):space+dot1+dot2+dot7",),
+			"title": ("br(braillePen):space+dot2+dot3+dot4+dot5",),
+			"reportCurrentFocus": ("br(braillePen):space+dot1+dot2+dot4+dot7",),
+			"reportStatusLine": ("br(braillePen):space+dot2+dot3+dot4+dot7",),
+			"speakForeground": ("br(braillePen):space+dot1+dot2+dot4+dot5+dot7",),
+			"kb:control+leftArrow": ("br(braillePen):space+dot2",),
+			"kb:control+rightArrow": ("br(braillePen):space+dot5",),
+			"kb:control+home": ("br(braillePen):space+dot2+dot3+dot7",),
+			"kb:control+end": ("br(braillePen):space+dot5+dot6+dot7",),
+		},
+	})
+
+class InputGesture(braille.BrailleDisplayGesture, brailleInput.BrailleInputGesture):
+
+	source = BrailleDisplayDriver.name
+
+	def __init__(self, brailleKeys,controlKeys):
+		super(InputGesture, self).__init__()
+		spaceKey=False
+		brailleKeysExtended = ord(brailleKeys)
+		controlKeysExtended = ord(controlKeys)
+		if controlKeysExtended & 1 >0:
+			spaceKey=True
```

Comment:
nit: You could simplify this to the following (removing the earlier `spaceKey = False`):

```
spaceKey = controlKeysExtended & 1
```